src/term/mod.rs

```diff
@@ -277,6 +290,25 @@ impl Term {
         fst.subst(from, to);
         snd.subst(from, to);
       }
+      Term::Tup { fst, snd } => {
+        fst.subst(from, to);
+        snd.subst(from, to);
+      }
+    }
+  }
+}
+
+impl LetPat {
```

Comment:
You can make this into a fmt::Display implementation (clippy will probably report this as well)